users/users_router.js

```diff
@@ -1,26 +1,36 @@
-const router = require('express').Router();
-const Users = require('./users_model');
+const router = require("express").Router();
+const Users = require("./users_model");
+const restricted = require("../middleware/restricted_middleware");
 
-router.get('/', async (req, res) => {
-    try {
-        const users = await Users.find();
-        res.json(users);
-    }
-    catch (err) {
-        res.send({ message: "Try again later.", err })
-    };
+router.get("/", restricted, async (req, res) => {
+  try {
+    const users = await Users.find();
+    res.json(users);
+  } catch (err) {
+    res.send({ message: "Try again later.", err });
+  }
 });
 
-router.get('/:id', async(req, res) => {
-    const { id } = req.params;
+router.get("/:id", async (req, res) => {
+  const { id } = req.params;
 
-    try {
-        const user = await Users.findById(id);
-        res.json(user);
-    }
-    catch (err) {
-        res.send({ message: "Try again later.", err })
-    };
+  try {
+    const user = await Users.findById(id);
+    res.json(user);
+  } catch (err) {
+    res.send({ message: "Try again later.", err });
+  }
 });
 
-module.exports = router;
\ No newline at end of file
+router.get("/username/:username", async (req, res) => {
+  const { username } = req.params;
+
+  try {
+    const id = await Users.findIdByUsername(username);
+    res.json(id);
+  } catch (err) {
+    res.send({ message: "Try again later.", err });
```

Comment:
We need to get some status codes in here for
 - 200 success
 - 400  error in request
 - 500  error on server

```suggestion
    res.status(500).send({ message: "Try again later.", err });
```